pkg/cloudprovider/cloudprovider.go

```diff
@@ -318,6 +278,111 @@ func (c *CloudProvider) RepairPolicies() []cloudprovider.RepairPolicy {
 	return []cloudprovider.RepairPolicy{}
 }
 
+func (c *CloudProvider) createOrGetMachine(ctx context.Context, nodeClaim *karpv1.NodeClaim) (*capiv1beta1.MachineDeployment, *capiv1beta1.Machine, error) {
```

Comment:
Nothing to do with the implementation, but I think this might benefit from either a better name or a well worded comment above the function definition.

I think it's a little confusing to name this `createOrGetMachine` because it might give the impression that the function is calling some sort of caching mechanism where we create a Machine, if it doesn't already exist. But actually, the "get" is supposed to act as a status check for the asynchronous provisioning.